crypto/err/err.c

```diff
@@ -228,8 +228,8 @@ static void build_SYS_str_reasons(void)
 
                 str->string = cur;
                 cnt += l;
-                if (cnt > sizeof(strerror_pool))
-                    cnt = sizeof(strerror_pool);
+                if (cnt >= sizeof(strerror_pool))
+                    cnt = sizeof(strerror_pool) - 1;
```

Comment:
note that I think the loop below does not handle the case properly
when the first string is empty, or contains only spaces.

Reply:
I agree, there should be a guard that ensures that cur doesn't leave the strerror_pool, such as
```  while (ossl_isspace(cur[-1]) && cur > strerror_pool) {```

Reply:
yes, just put the condition the other way round.